api/package.json

```
@@ -135,6 +137,7 @@
     "test:api:watch": "NODE_ENV=test mocha --recursive tests --watch --reporter dot",
     "test:api:bail": "npm run test:api:unit -- --bail && npm run test:api:integration -- --bail && npm run test:api:acceptance -- --bail",
     "test:ci": "npm run test",
+    "test:database": "schemalint --config tests/acceptance/database/configuration.js",
```

Si c'est du lint c'est plutôt une tâche à déplacer dans le lint non ? 

```suggestion
    "lint:database": "schemalint --config tests/acceptance/database/configuration.js",
```

---

Pas bien compris pourquoi c'est du `test` et pas du `lint` également.

Pour aller plus loin : si ce script est stateful, est-ce qu'on ne devrait pas lancer un `db:reset` automatiquement avant ce "lint" pour garantir l'état de la base ?

---

Héhé c'est expliqué dans la PR. Dites-moi si c'est pas clair

>Le schéma de BDD à un instant t est la résultante de l'application de toutes les migrations, il faut donc une solution stateful. La seule solution connu est de linter le schéma de BDD instancié.

---

Non pour moi c'est pas clair que `test` = BBD instancié, et `lint` = BDD pas instanciée (si j'ai bien compris)

---

ahhhh je crois deviner
C'est Ok pour toi si on instancie la BDD puis qu'on lance lint:bdd après ?
La solution que j'ai prise était pour éviter de créer le schéma 2 fois, mais c'est pas si important si c'est plus lisible.

---

Ah c'est de l'optim parce que les tests créént le schema, c'est ça ? Perso je préfère vraiment `lint` même si ça nécessite d'avoir le schéma en prérequis.

---

Tu as complètement raison, je fais ça de suite

---

Pour la reproductibilité : est-ce qu'il faut pas `db:reset` systématiquement ?

---

`db:prepare` plutôt, on a pas besoin de seeds en test

---

C'est un impact à bien noter si on prend cette direction mais IMO je préfère largement la reproductibilité !

---

Ah mais c'était pas négociable je te rassure !

---

D'ailleurs je ne suis pas bien sûr si on peut linter la BDD de test avec `NODE_ENV=test` ? Pour éviter de drop sa BDD de dev

---

C'est l'option que j'ai prise, car c'est la plus simple sur CircleCI